### PR TITLE
Initial commit for zsh native completion support

### PR DESCRIPTION
**Pull request by camfrout**
### Add support for native zsh completion for the dbt command

As mentioned in issue #6:

**Pros:**

- Completion for models works with strings in the middle of the model names (e.g. typing invoice will match models customer_invoice and vendor_invoice
- Completion has been set up for some flags and commands (listing most of the flags available when running dbt run, dbt test and dbt docs as well as the global flags)
- Support for '+' and '@' has been replicated from the bash completion script
- [NEW] This script supports separating models with a comma, as defined in dbt 0.18

**Cons:**

- Currently, when running dbt run or dbt test the models will be suggested even though there is no model selection flag defined or if the model selection flag is not the last one in the list

### CODE REVIEW

**Comment by grantwinship** [comment]
@camfrout could you include proposed README updates (if any) for installation process if taking this route? i'd love to test this out but i'm not super familiar with completion functionality in zsh and don't want to accidentally bork up my dotfiles making guesses 😅

**Comment by camfrout** [comment]
Sure, I have just added instructions. The easy way is if you are already using oh-my-zsh.

File: README.md [hunk 1]
```diff
@@ -36,6 +36,23 @@ echo 'autoload -U +X bashcompinit && bashcompinit' >> ~/.zshrc
...
+
+#### Using oh-my-zsh
+
+If using oh-my-zsh, the atutocompletion script can be installed by putting the `_dbt` file in the folder `~/.oh-my-zsh/completions/` (create the completions folder if it doesn't exist).
+
+#### Manual installation
+
+If installing manually, instructions from [zsh-completions](https://github.com/zsh-users/zsh-completions) mention that you need to include the directory where `_dbt` is stored in your `$fpath` by adding the following to `~/.zshrc`
+```
+fpath=(path/to/_dbt/parent_direcotry $fpath)
```
Comment (grantwinship) [inline README.md:49]:
```suggestion
fpath=(path/to/_dbt/parent_directory $fpath)
```

**Comment by grantwinship** [comment]
> Sure, I have just added instructions. The easy way is if you are already using oh-my-zsh.

Awesome, thank you! I've got it up and running, so far working very nicely! I'll put it through its paces on client work tomorrow - thanks again this is very cool 🌻

**Comment by grantwinship** [comment]
i've had zero issues with this @jtcohen6 -- functions flawlessly, install was a breeze: 👍🏻 👍🏻 from me to merge

**Comment by jtcohen6** [comment]
Thanks for testing @grantwinship!

@camfrout Tiny thing: could you rename the file from `_dbt` to something like `dbt-completion.zsh`, to mirror the existing bash version?

**Comment by camfrout** [comment]
Hi.

I don't think I can, it is actually part of the naming convention  for ZSH completion scripts.

[Source](https://github.com/zsh-users/zsh-completions/blob/master/zsh-completions-howto.org#telling-zsh-which-function-to-use-for-completing-a-command): "_Completion functions for commands are stored in files with names beginning with an underscore \_, and these files should be placed in a directory listed in the $fpath variable. You can add a directory to $fpath by adding a line like this to your ~/.zshrc file:_"

Looking at [this repo](https://github.com/zsh-users/zsh-completions/tree/master/src) of ZSH completion scripts, we can see that all the scripts are named `_<command>`

**Comment by jtcohen6** [comment]
TIL! Thanks for the links @camfrout. I'm good with merging this as-is, then :)

**Pull request merged**